runtime/src/message_processor.rs

```diff
@@ -677,36 +681,43 @@ impl MessageProcessor {
     pub fn native_invoke(
         invoke_context: &mut dyn InvokeContext,
         instruction: Instruction,
-        keyed_accounts: &[&KeyedAccount],
-        signers_seeds: &[&[&[u8]]],
+        keyed_account_indices: &[usize],
```

Comment:
If we passed `&[Rc<KeyedAccounts>]` would that allow us to avoid using indicies?

Reply:
The problem here is not that the lifetime or ownership is unclear, but that there could be multiple references of which at least one is mutable (the one in `&mut dyn InvokeContext`). So I think a runtime borrow check like `Cell` would be more appropriate here.

Reply:
But it is not necessary, that is why I used indices.

Reply:
Lets try to get rid of the indicies